framework/include/utils/RankTwoTensorImplementation.h

```diff
@@ -562,30 +597,25 @@ RankTwoTensorTempl<T>
 RankTwoTensorTempl<T>::deviatoric() const
 {
   RankTwoTensorTempl<T> deviatoric(*this);
-  deviatoric.addIa(-1.0 / 3.0 * this->tr()); // actually construct deviatoric part
+  // actually construct deviatoric part
+  deviatoric.addIa(-1.0 / 3.0 * this->tr());
   return deviatoric;
 }
 
 template <typename T>
 T
 RankTwoTensorTempl<T>::generalSecondInvariant() const
```

Comment:
I have some strong opinions about the names of `generalSecondInvariant` and `secondInvariant`, because there is no such thing called a general second invariant, there is only second invariant of a rank two tensor.

The `generalSecondInvariant` should be called `secondInvariant`, and the `secondInvariant` here is just a shortcut for `A.deviatoric().secondInvariant()`. I appreciate the savings in the computational cost using this shortcut, but I am not a fan of this name.

Reply:
Same thing applies to the `thirdInvariant` below...

Reply:
Anyways, not related to this PR.

Reply:
Noted in #21269